lib_eio/pool.mli

```diff
@@ -0,0 +1,24 @@
+type 'a t
+
+type 'a handlers = {
+  check : 'a -> bool;
+  dispose : 'a -> unit;
+}
+
+val noop_handlers : 'a handlers
+
+val create :
+  sw:Switch.t ->
+  alloc:(unit -> 'a * 'a handlers) ->
+  int ->
+  'a t
+
+val use : 'a t -> ('a -> 'b) -> 'b
+
+val async : sw:Switch.t -> 'a t -> ('a -> unit) -> unit
+
+val async_promise : sw:Switch.t -> 'a t -> ('a -> 'b) -> 'b Promise.or_exn
+
+val clear : 'a t -> unit
```

Comment:
I don't quite understand when `clear` would be used, this calls the `dispose` function for the resource, how would people use this in practice? I think a few test cases would help here.

Reply:
I think I just copied this from Lwt_pool. Indeed not very obvious how to use this in practice - I think removing it might be best.